lib/active_merchant/billing/gateways/payu_in.rb

```diff
@@ -0,0 +1,234 @@
+# encoding: utf-8
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayuInGateway < Gateway
+      self.test_url = "https://test.payu.in/_payment"
+      self.live_url = "https://secure.payu.in/_payment"
+
+      TEST_INFO_URL = "https://test.payu.in/merchant/postservice.php?form=2"
+      LIVE_INFO_URL = "https://info.payu.in/merchant/postservice.php?form=2"
+
+      self.supported_countries = ['IN']
+      self.default_currency = 'INR'
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club]
+
+      self.homepage_url = 'https://www.payu.in/'
+      self.display_name = 'PayU India'
+
+      def initialize(options={})
+        requires!(options, :key, :salt)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        requires!(options, :order_id)
+
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_addresses(post, options)
+        add_customer_data(post, options)
+        add_auth(post)
+
+        MultiResponse.run do |r|
+          r.process{commit(url("purchase"), post)}
+          if(r.params["enrolled"].to_s == "0")
+            r.process{commit(r.params["post_uri"], r.params["form_post_vars"])}
```

Comment:
So if they're not 3DS enrolled, we POST back to a URL they give us, using params they give us? Is there value in this flow I don't understand, or is it a silly API?